recipes/libsolace/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from conans import CMake, ConanFile, tools
+
+
+class LibsolaceConan(ConanFile):
+    name = "libsolace"
+    description = "High performance components for mission critical applications"
+    topics = ("HPC", "High reliability", "P10", "solace", "performance", "c++", "conan")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/abbyssoul/libsolace"
+    license = "Apache-2.0"
+    author = "Ivan Ryabov <abbyssoul@gmail.com>"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = "shared=False", "fPIC=True"
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        # Copy license file
+        cmake = CMake(self, parallel=True)
+        cmake.definitions["BUILD_TESTING"] = "OFF"
+        cmake.definitions["PKG_CONFIG"] = "OFF"
+        cmake.definitions["SOLACE_GTEST_SUPPORT"] = "OFF"
+        cmake.configure(source_folder=self._source_subfolder)
+        cmake.install()
```

Comment:
Hm.. my understanding was that cmake.install() will just create a local install into a build directory that gets packaged.

Reply:
Yep - just tested it - that was intential. So with cmake.buld() the package is built but not installed and so is not available to test package. That is when a test package gets built - it can not locate the library.

Reply:
`install()` does a local install in the package folder, so you can reuse the CMake install logic and omit the `package()` step. However, before doing that, you should build the library (in case it is not a header-only library)

Reply:
Not sure I understood it. calling cmake.install() does imply cmake.build() as `install` step depends on artifacts being built. 
So - what you are suggesting is that in conan build step I can replace cmake.install() with cmake.build() - and call `cmake.install()` from `package()`. Is that currect?

Reply:
Correct 😄 That is very common and documented here: https://docs.conan.io/en/latest/howtos/cmake_install.html

You can take the Poco library recipe as an example: https://github.com/conan-io/conan-center-index/blob/master/recipes/poco/all/conanfile.py#L124

Reply:
Ok - I did a bit of refactoring to re-use `cmake.install` form `package` step. Seems to work ok.

Reply:
Looking good! 🎉 thanks!